service/assessment/assessment.go

```diff
@@ -166,56 +165,69 @@
 		s.informHooks(nil, newError)
 
 		res = &assessment.AssessEvidenceResponse{
-			Status: false,
+			Status:        assessment.AssessEvidenceResponse_FAILED,
+			StatusMessage: newError.Error(),
 		}
 
-		return res, status.Errorf(codes.InvalidArgument, "invalid req: %v", newError)
+		return res, status.Error(codes.InvalidArgument, "invalid evidence")
 	}
 
 	// Assess evidence
 	err = s.handleEvidence(req.Evidence, resourceId)
 
 	if err != nil {
 		res = &assessment.AssessEvidenceResponse{
-			Status: false,
+			Status:        assessment.AssessEvidenceResponse_FAILED,
+			StatusMessage: err.Error(),
 		}
 
-		return res, status.Errorf(codes.Internal, "error while handling evidence: %v", err)
+		log.Errorf("Error while handling evidence: %v", err)
+
+		return res, status.Error(codes.Internal, "error while handling evidence")
 	}
 
 	res = &assessment.AssessEvidenceResponse{
-		Status: true,
+		Status: assessment.AssessEvidenceResponse_ASSESSED,
 	}
 
 	return res, nil
 }
 
-// AssessEvidences is a method implementation of the assessment interface: It assesses multiple evidences (stream)
+// AssessEvidences is a method implementation of the assessment interface: It assesses multiple evidences (stream) and responds with a stream.
 func (s *Service) AssessEvidences(stream assessment.Assessment_AssessEvidencesServer) (err error) {
 	var (
 		req *assessment.AssessEvidenceRequest
+		res *assessment.AssessEvidenceResponse
 	)
 
 	for {
+
+		// TODO(all): Check context?
 		req, err = stream.Recv()
 
+		// If no more input of the stream is available, return
+		if err == io.EOF {
+			return nil
+		}
 		if err != nil {
-			// If no more input of the stream is available, return SendAndClose `error`
-			if err == io.EOF {
-				log.Infof("Stopped receiving streamed evidences")
-				return stream.SendAndClose(&emptypb.Empty{})
-			}
-
-			return err
+			log.Errorf("Assessment: Cannot receive stream request: %v", err)
```

Comment:
```suggestion
			log.Errorf("Cannot receive stream request: %v", err)
```

Reply:
If omitted, also consider the following structure to avoid duplicated strings:
```
			err = fmt.Errorf("cannot receive stream request: %v", err)
			log.Error(err)
			return status.Error(codes.Unknown, err.Error())
```

Reply:
The "problem" here is, that we usually capitalize the log messages and return errors in lower case. But we could also do it like that
`
err = fmt.Errorf("cannot receive stream request: %v", err)
log.Error(strings.Title(err.Error()))
return status.Error(codes.Unknown, err.Error())
`